File: src/main/java/gabrielruiu/packt/scheduler/EmailSubjectResolver.java
package gabrielruiu.packt.scheduler;

import org.springframework.stereotype.Component;

import java.text.SimpleDateFormat;
import java.util.Date;

/**
 * @author Gabriel Mihai Ruiu (gabriel.ruiu@mail.com)
 */
@Component
public class EmailSubjectResolver {

    public String determineEmailSubject(String bookTitle) {
        return bookTitle + " [" + getCurrentDate() + "]";
    }

    private String getCurrentDate() {
        SimpleDateFormat sdf = new SimpleDateFormat("dd.MM.yyy");
        return sdf.format(new Date());
    }
}

File: src/main/kotlin/gabrielruiu/packt/scheduler/EmailSubjectResolver.kt
package gabrielruiu.packt.scheduler

import org.springframework.stereotype.Component
import java.text.SimpleDateFormat
import java.util.*

@Component
open class EmailSubjectResolver {

    fun determineEmailSubject(bookTitle: String): String {
        return bookTitle + '[' + getCurrentDate() + ']'
    }

    private fun getCurrentDate(): String {
        val sdf = SimpleDateFormat("dd.MM.yyyy")
        return sdf.format(Date())
    }
}
